src/main/java/domain/ladder/Ladder.java

```diff
@@ -32,20 +32,55 @@ private List<LadderLeg> generateFirstToLastLegs(DirectionGenerator directionGene
         return generatedLadderLegs;
     }
 
-    public List<Direction> getDirectionAtHorizontalIndex(int index) {
-        validateIndex(index);
+
+    public int moveCoordinateToResultPoint(int x, int y) {
+        if (isReachResultPoint(y)) return x;
+        Direction currentDirection = getDirectionOfLadderLegPieceAtSpecificCoordinate(x, y);
+        if (currentDirection.equals(Direction.RIGHT)) {
+            x += 1;
+        }
+        if (currentDirection.equals(Direction.LEFT)) {
+            x -= 1;
+        }
+        y += 1;
+        return moveCoordinateToResultPoint(x, y);
```

Comment:
```suggestion
        if (currentDirection.equals(Direction.RIGHT)) {
            return moveCoordinateToResultPoint(x + 1, y + 1);
        }
        if (currentDirection.equals(Direction.LEFT)) {
            return moveCoordinateToResultPoint(x - 1, y + 1);
        }
        return moveCoordinateToResultPoint(x, y + 1);
```
현재의 경우에는 무엇이 더해졌는지 기억한 다음 46라인 리턴문을 보아야하기 때문에 읽는데 어려움이 있을 수도 있어 위와 같이 표현되는게 어떨까해요.